### PR TITLE
IA-2529: Completeness stats table use legend color

### PR DESCRIPTION
**Pull request by beygorghor**
Make completeness stats table color coded with legend thresholds

Related JIRA tickets : IA-2529

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [x] Are there enough tests
- [x] Documentation has been included (for new feature)

## Changes

- Update front-end guidelines with `sx` use 
- split completeness column and use new `DescendantsCell`, `ItselfCell` to render stats
- `useEffectiveThreshold` to get form threshold or default one
- display legend while hovering cell
- modify api to return `legend_threshold`
- fix python tests

## How to test

You need forms linked to org units to display completeness stats values
Go to the page and navigate through the org units

## Print screen / video

https://github.com/BLSQ/iaso/assets/12494624/3206244f-0321-430e-bed6-b6392cf3c205

https://github.com/BLSQ/iaso/assets/12494624/9e2bd117-6fe2-4dfb-ba53-3708d7e8df70

### CODE REVIEW

**Review by mathvdh — APPROVED** [review]
Ok for me on the backend side. These SQL queries are complicated ;)

File: docs/pages/dev/reference/guidelines/front-end/front-end.md
Comment (quang-le) [inline docs/pages/dev/reference/guidelines/front-end/front-end.md:1]: This should have been a separate DOC PR

Comment (quang-le) [reply docs/pages/dev/reference/guidelines/front-end/front-end.md:1]: but ❤️

File: hat/assets/js/apps/Iaso/domains/completenessStats/hooks/useEffectiveThreshold.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+const defaultScaleThreshold = {
+    domain: [70, 90],
+    range: ['red', 'orange', 'green'],
+};
+
+export const useEffectiveThreshold = (
+    threshold?: ScaleThreshold,
+): ScaleThreshold =>
+    !threshold || isEqual(threshold, {}) ? defaultScaleThreshold : threshold;
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/completenessStats/hooks/useEffectiveThreshold.ts]: In my opinion, if it has no internal state, it shouldn't be a hook, just a normal function

**Review by quang-le — CHANGES_REQUESTED** [review]
Works very nicely. Just a small block on `useEffectiveThreshold`. I'll update the guidelines to document it

**Review by quang-le — APPROVED** [review]

**Pull request merged**